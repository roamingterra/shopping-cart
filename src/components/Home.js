import React from "react";
import { Link } from "react-router-dom";
import homeCss from "./../styles/home.module.css";

function Home() {
  function blackPageBackground() {
    const html = document.querySelector("html");
    const body = document.querySelector("body");
    html.style.backgroundColor = "black";
    body.style.backgroundColor = "black";
  }

  blackPageBackground();

  return (
    <div className={homeCss["home-component"]} key={"home"}>
      <div className={homeCss["content-wrapper"]}>
        <div className={homeCss["main-content-1"]}>
          <div className={homeCss["header"]}>
            <div className={homeCss["logo-large"]}>roamingterra</div>
            <div className={homeCss["sign-in"]}>sign in</div>
          </div>
          <div className={homeCss["container-1"]}>
            <div className={homeCss["slogan"]}>
              <div>Make the most out of this world</div>
            </div>
            <div className={homeCss["slogan-elaboration"]}>
              Learn some of the world's most spoken languages at home, and on
              the go
            </div>
            <Link to="/shopping">
              <button className={homeCss["shop-1"]}>
                <span>Shop</span>
              </button>
            </Link>
          </div>
        </div>
        <div className={homeCss["main-content-2"]}>
          <div className={homeCss["description-container-2"]}>
            <div className={homeCss["key-word-2"]}>VOCABULARY BUILDING</div>
            <div className={homeCss["title-2"]}>
              Expand your language skills
            </div>
            <div className={homeCss["description-2"]}>
              Build a robust vocabulary with an extensive library of words and
              phrases, and practice vocabulary through interactive exercises and
              immersive activities. <br></br>
              <br></br>Master essential words and idiomatic expressions for
              real-world conversations with context-based examples and usage
              tips.
            </div>
          </div>
          <img
            className={homeCss["london"]}
            src="./images/london-night.jpeg"
            alt="london at night"
          ></img>
        </div>
        <div className={homeCss["main-content-3"]}>
          <img
            className={homeCss["paris"]}
            src="./images/paris-night.jpeg"
            alt="paris at night"
          ></img>
          <div className={homeCss["description-container-3"]}>
            <div className={homeCss["key-word-3"]}>INTERACTIVE LESSONS</div>
            <div className={homeCss["title-3"]}>Engage in dynamic learning</div>
            <div className={homeCss["description-3"]}>
              Experience interactive lessons that keep you engaged and
              motivated. Immerse yourself in multimedia content, including
              audio, video, and interactive exercises. <br></br>
              <br></br>Practice speaking and listening skills with speech
              recognition technology. Receive instant feedback and personalized
              recommendations for improvement.
            </div>
          </div>
        </div>
        <div className={homeCss["main-content-4"]}>
          <div className={homeCss["description-container-4"]}>
            <div className={homeCss["key-word-4"]}>CULTURAL IMMERSION</div>
            <div className={homeCss["title-4"]}>
              Explore language and culture
            </div>
            <div className={homeCss["description-4"]}>
              Immerse yourself in the richness of different cultures through
              authentic content. Discover customs, traditions, and cultural
              nuances while learning the language.<br></br>
              <br></br> Engage in dialogues and scenarios that reflect real-life
              cultural situations and gain a deeper understanding of the
              language's cultural context.
            </div>
          </div>

          <img
            className={homeCss["japan"]}
            src="./images/japan-night.jpeg"
            alt="japan at night"
          ></img>
        </div>
        <div className={homeCss["main-content-5"]}>
          <div className={homeCss["description-container-5"]}>
            <div className={homeCss["key-word-5"]}>Choose your language</div>
            <div className={homeCss["description-5"]}>
              Unlock the world, discover your language journey
            </div>
            <Link to="/shopping">
              <button className={homeCss["shop-2"]}>
                <span>Shop</span>
              </button>
            </Link>
          </div>
        </div>
        <div className={homeCss["main-content-6"]}>
          <div>Have a question?</div>
          <div className={homeCss["cards-container"]}>
            <div className={homeCss["help-center"]}>
              <div className={homeCss["help-center-text"]}>
                <div className={homeCss["help-center-text-title"]}>
                  Go to Help Center
                </div>
                <div className={homeCss["help-center-text-description"]}>
                  Find answers and suggestions to most common questions
                </div>
              </div>
              <img src="./images/help-center.png" alt="help center icon"></img>
            </div>
            <div className={homeCss["community"]}>
              <div className={homeCss["community-text"]}>
                <div className={homeCss["community-text-title"]}>
                  Ask the community
                </div>
                <div className={homeCss["community-text-description"]}>
                  Create connections with music creators like you
                </div>
              </div>
              <img src="./images/community.png" alt="community icon"></img>
            </div>
          </div>
        </div>
        <div className={homeCss["info"]}>
          <div className={homeCss["logo"]}>
            <div className={homeCss["logo-wrapper"]}>
              <img src="./images/earth-logo.png" alt="earth logo" />
              <span>roamingterra</span>
            </div>
          </div>
          <div className={homeCss["company-info-container"]}>
            <div>Company</div>
            <div>About us</div>
            <div>People</div>
            <div>Brand Resources</div>
            <div>Press</div>
            <div>Contact</div>
            <div>Blog</div>
            <div>Podcast</div>
          </div>
          <div className={homeCss["products-info-container"]}>
            <div>Products</div>
            <div>For Teachers</div>
            <div>For Schools</div>
            <div>For Partners</div>
            <div>For Developers</div>
            <div>For the Community</div>
          </div>
          <div className={homeCss["community-info-container"]}>
            <div>Community</div>
            <div>Overview</div>
            <div>Guidelines</div>
            <div>Become a Curator</div>
            <div>Help Center</div>
            <div>Ask the Community</div>
            <div>Community Blog</div>
          </div>
        </div>
      </div>
      <div className={homeCss["footer"]}>
        <div className={homeCss["logo-small"]}>roamingterra</div>
        <div className={homeCss["legal-information"]}>
          <div>Privacy Policy</div>
          <div>Cookie Policy</div>
          <div>EULA</div>
          <div>Copyright</div>
        </div>
        <div className={homeCss["credits"]}>
          <div>Website created by Daniel Ballerini</div>
          <div>
            with help from{" "}
            <a href="https://www.theodinproject.com/"> Â© Odin Project</a>
          </div>
        </div>
      </div>
    </div>
  );
}

export default Home;
