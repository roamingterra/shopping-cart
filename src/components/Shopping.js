import React from "react";
import Icon from "@mdi/react";
import { mdiCartOutline } from "@mdi/js";
import { mdiWindowClose } from "@mdi/js";
import Product from "./Product";
import ShoppingCartProduct from "./ShoppingCartProduct";
import shoppingCss from "./../styles/shopping.module.css";
import { Link } from "react-router-dom";

function Shopping() {
  const productInfo = {
    productNames: {
      english: "English",
      french: "French",
      mandarin: "Mandarin",
      spanish: "Spanish",
      german: "German",
      japanese: "Japanese",
      italian: "Italian",
      portuguese: "Portuguese",
      hindi: "Hindi",
      arabic: "Arabic",
    },
    imagePaths: {
      english: "/images/united-states-flag.png",
      french: "/images/france-flag.png",
      mandarin: "/images/china-flag.png",
      spanish: "/images/spain-flag.png",
      german: "/images/germany-flag.png",
      japanese: "/images/japan-flag.png",
      italian: "/images/italy-flag.png",
      portuguese: "/images/brazil-flag.webp",
      hindi: "/images/india-flag.png",
      arabic: "/images/lebanon-flag.png",
    },
    descriptions: {
      english: [
        "Widely spoken worldwide",
        "Language of business, travel, and academia",
        "Immerse yourself in English literature and entertainment",
      ],
      french: [
        "Dive into the language of love and sophistication",
        "Experience the elegance of French cuisine, fashion, and arts",
        "Expand your career prospects with fluency in the language of diplomacy",
      ],
      mandarin: [
        "Spoken by over a billion people",
        "Vibrant culture, profound history, and booming business opportunities",
        "Gain an edge in an increasingly interconnected world",
      ],
      spanish: [
        "Immerse yourself in the passion and vibrancy of the Spanish-speaking world",
        "Explore diverse cultures across Spain and Latin America",
        "Expand your horizons and connect with millions of Spanish speakers",
      ],
      german: [
        "Discover the language of precision, engineering, and innovation",
        "Immerse yourself in Germany's rich cultural heritage",
        "Career opportunities in industries such as automotive and technology",
      ],
      japanese: [
        "Japan Blends tradition with technological prowess",
        "Dive into Japanese pop culture",
        "One of Asia's economic powerhouses",
      ],
      italian: [
        "Indulge in the language of art, music, and exquisite cuisine",
        "Discover Italy's historic cities, breathtaking landscapes, and renowned fashion",
        "Unlock the treasures of Italy's rich cultural heritage",
      ],
      portuguese: [
        "The language of Brazil and Portugal",
        "Explore diverse ecosystems and vibrant festivals of Portuguese-speaking countries",
        "Connect with millions of Portuguese speakers across continents",
      ],
      hindi: [
        "Dive into the colorful world of Bollywood, music, and spirituality",
        "Explore the diverse cultures and traditions of India",
        "Connect with a billion Hindi speakers worldwide",
      ],
      arabic: [
        "Embark on a journey to the cradle of civilization",
        "Discover the poetic nature of Arabic script",
        "Gain a competitive edge in diplomacy and business in the Middle East and North Africa",
      ],
    },
    pricing: {
      english: 99.99,
      french: 79.99,
      mandarin: 109.99,
      spanish: 89.99,
      german: 79.99,
      japanese: 99.99,
      italian: 79.99,
      portuguese: 79.99,
      hindi: 89.99,
      arabic: 99.99,
    },
  };

  function whitePageBackground() {
    const html = document.querySelector("html");
    const body = document.querySelector("body");
    html.style.backgroundColor = "white";
    body.style.backgroundColor = "white";
  }

  whitePageBackground();

  return (
    <div
      data-testid="shopping-page"
      className={shoppingCss["shopping-component"]}
    >
      <div className={shoppingCss["side-bar"]}>
        <div className={shoppingCss["side-bar-header"]}>
          <Icon
            className={shoppingCss["side-bar-close"]}
            path={mdiWindowClose}
            size={2}
          />
          <Icon
            className={shoppingCss["side-bar-shopping-cart"]}
            path={mdiCartOutline}
            title="Shopping Cart"
            size={2}
          />
        </div>
        <div className={shoppingCss["shopping-cart-empty-message"]}>
          Your cart is empty
        </div>
        <div className={shoppingCss["side-bar-main-content"]}>
          <div className={shoppingCss["side-bar-products"]}>
            {/* <ShoppingCartProduct
              productName={productInfo.productNames.english}
              image={productInfo.imagePaths.english}
              price={productInfo.pricing.english}
            ></ShoppingCartProduct> */}
            {/* <ShoppingCartProduct
              productName={productInfo.productNames.french}
              image={productInfo.imagePaths.french}
              price={productInfo.pricing.french}
            ></ShoppingCartProduct> */}
            {/* <ShoppingCartProduct
              productName={productInfo.productNames.arabic}
              image={productInfo.imagePaths.arabic}
              price={productInfo.pricing.arabic}
            ></ShoppingCartProduct> */}
            {/* <ShoppingCartProduct
              productName={productInfo.productNames.spanish}
              image={productInfo.imagePaths.spanish}
              price={productInfo.pricing.spanish}
            ></ShoppingCartProduct> */}
            {/* <ShoppingCartProduct
              productName={productInfo.productNames.mandarin}
              image={productInfo.imagePaths.mandarin}
              price={productInfo.pricing.mandarin}
            ></ShoppingCartProduct> */}
          </div>
          <div className={shoppingCss["side-bar-sub-total"]}>
            <div>Subtotal</div>
            <div className={shoppingCss["total"]}>$100</div>
          </div>
          <button className={shoppingCss["side-bar-checkout"]}>
            PROCEED TO CHECKOUT
          </button>
        </div>
      </div>
      <div className={shoppingCss["overlay"]}></div>
      <div className={shoppingCss["content-and-footer-wrapper"]}>
        <div className={shoppingCss["content-wrapper"]}>
          <div className={shoppingCss["header"]}>
            <Link to="/">
              <div className={shoppingCss["logo-large"]}>
                <div>roamingterra</div>
              </div>
            </Link>
            <Icon
              className={shoppingCss["shopping-cart"]}
              path={mdiCartOutline}
              title="Shopping Cart"
              size={2}
            />
          </div>
          <div className={shoppingCss["main-content-1"]}>
            Choose your language learning software
          </div>
          <div className={shoppingCss["main-content-2"]}>
            <Product
              productName={productInfo.productNames.english}
              image={productInfo.imagePaths.english}
              description={productInfo.descriptions.english}
              price={productInfo.pricing.english}
            ></Product>
            <Product
              productName={productInfo.productNames.french}
              image={productInfo.imagePaths.french}
              description={productInfo.descriptions.french}
              price={productInfo.pricing.french}
            ></Product>
            <Product
              productName={productInfo.productNames.mandarin}
              image={productInfo.imagePaths.mandarin}
              description={productInfo.descriptions.mandarin}
              price={productInfo.pricing.mandarin}
            ></Product>
            <Product
              productName={productInfo.productNames.spanish}
              image={productInfo.imagePaths.spanish}
              description={productInfo.descriptions.spanish}
              price={productInfo.pricing.spanish}
            ></Product>
            <Product
              productName={productInfo.productNames.german}
              image={productInfo.imagePaths.german}
              description={productInfo.descriptions.german}
              price={productInfo.pricing.german}
            ></Product>
            <Product
              productName={productInfo.productNames.japanese}
              image={productInfo.imagePaths.japanese}
              description={productInfo.descriptions.japanese}
              price={productInfo.pricing.japanese}
            ></Product>
            <Product
              productName={productInfo.productNames.italian}
              image={productInfo.imagePaths.italian}
              description={productInfo.descriptions.italian}
              price={productInfo.pricing.italian}
            ></Product>
            <Product
              productName={productInfo.productNames.portuguese}
              image={productInfo.imagePaths.portuguese}
              description={productInfo.descriptions.portuguese}
              price={productInfo.pricing.portuguese}
            ></Product>
            <Product
              productName={productInfo.productNames.hindi}
              image={productInfo.imagePaths.hindi}
              description={productInfo.descriptions.hindi}
              price={productInfo.pricing.hindi}
            ></Product>
            <Product
              productName={productInfo.productNames.arabic}
              image={productInfo.imagePaths.arabic}
              description={productInfo.descriptions.arabic}
              price={productInfo.pricing.arabic}
            ></Product>
          </div>
          <div className={shoppingCss["main-content-3"]}>
            <div className={shoppingCss["main-content-3-title"]}>FAQs</div>
            <div className={shoppingCss["frequently-asked-questions"]}>
              <div className={shoppingCss["faq"]}>
                <div>
                  <div>What are the cancellation terms?</div>
                </div>
                <img src="/images/arrow-icon.svg" alt="arrow icon"></img>
              </div>
              <div className={shoppingCss["faq"]}>
                <div>
                  <div>
                    How long does it take to become fluent in a new language
                    using this software?
                  </div>
                </div>
                <img src="/images/arrow-icon.svg" alt="arrow icon"></img>
              </div>
              <div className={shoppingCss["faq"]}>
                <div>
                  <div>
                    Can I access the language learning materials offline?
                  </div>
                </div>
                <img src="/images/arrow-icon.svg" alt="arrow icon"></img>
              </div>
            </div>
          </div>
          <div className={shoppingCss["main-content-4"]}>
            <div>Let's talk</div>
            <div>
              Do you have any questions? Contact us and let's discuss it.
            </div>
            <button>Contact us</button>
          </div>
        </div>
        <div className={shoppingCss["footer"]}>
          <Link to="/" style={{ textDecoration: "none" }}>
            <div className={shoppingCss["logo-small"]}>roamingterra</div>
          </Link>
          <div className={shoppingCss["legal-information"]}>
            <div>Privacy Policy</div>
            <div>Cookie Policy</div>
            <div>EULA</div>
            <div>Copyright</div>
          </div>
          <div className={shoppingCss["credits"]}>
            <div>Website created by Daniel Ballerini</div>
            <div>
              with help from{" "}
              <a href="https://www.theodinproject.com/"> © Odin Project</a>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
}

export default Shopping;
