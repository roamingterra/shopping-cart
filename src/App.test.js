import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import { MemoryRouter } from "react-router-dom";
import Home from "./components/Home";
import Shopping from "./components/Shopping";

// ROUTE TESTING
describe("Test Routing Works", () => {
  // Test the web app starting at the home page
  test("Initial navigation to the home page", () => {
    const { asFragment } = render(
      // asFragment returns a serializable representation of the rendered component's output.
      <MemoryRouter>
        <Home />
      </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });
  // Test the routing behavior from the home page to the shopping page
  test("Navigate to the shopping page", () => {
    render(
      // Memory Router simulates routing behavior
      <MemoryRouter initialEntries={["/"]}>
        <Home />
        <Shopping />
      </MemoryRouter>
    );

    // Get all the buttons with the text "shopping"
    const shoppingButtons = screen.getAllByText("Shop");

    // Assert that there are multiple buttons
    expect(shoppingButtons).toHaveLength(2);

    // Iterate over the buttons and simulate a click event
    shoppingButtons.forEach((button) => {
      fireEvent.click(button);

      // Assert that the navigation occurred
      const shoppingPage = screen.getByTestId("shopping-page");
      expect(shoppingPage).toBeInTheDocument();
    });
  });
});

// LOGIC TESTING
// Test the functionality of the input fields for the user to manually type in the item quantities

// Test the functionality of the increment and decrement buttons for the item quantities

// Test the functionality of the add to cart buttons

// Test adding multiple items to the shopping cart and see if the cart correctly displays the total...
// quantity and the total price

// ACCESSIBILITY TESTING ???
// Perform accessibility tests to ensure the components are accessible and responsive across various devices
