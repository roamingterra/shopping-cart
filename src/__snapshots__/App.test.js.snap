// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Routing Works Initial navigation to the home page 1`] = `
<DocumentFragment>
  <div
    class="home-component"
  >
    <div
      class="content-wrapper"
    >
      <div
        class="main-content-1"
      >
        <div
          class="header"
        >
          <div
            class="logo-large"
          >
            roamingterra
          </div>
          <div
            class="sign-in"
          >
            sign in
          </div>
        </div>
        <div
          class="container-1"
        >
          <div
            class="slogan"
          >
            <div>
              Make the most out of this world
            </div>
          </div>
          <div
            class="slogan-elaboration"
          >
            Learn some of the world's most spoken languages at home, and on the go
          </div>
          <a
            href="/shopping"
          >
            <button
              class="shop-1"
            >
              <span>
                Shop
              </span>
            </button>
          </a>
        </div>
      </div>
      <div
        class="main-content-2"
      >
        <div
          class="description-container-2"
        >
          <div
            class="key-word-2"
          >
            VOCABULARY BUILDING
          </div>
          <div
            class="title-2"
          >
            Expand your language skills
          </div>
          <div
            class="description-2"
          >
            Build a robust vocabulary with an extensive library of words and phrases, and practice vocabulary through interactive exercises and immersive activities. 
            <br />
            <br />
            Master essential words and idiomatic expressions for real-world conversations with context-based examples and usage tips.
          </div>
        </div>
        <img
          alt="london at night"
          class="london"
          src="/images/london-night.jpeg"
        />
      </div>
      <div
        class="main-content-3"
      >
        <img
          alt="paris at night"
          class="paris"
          src="/images/paris-night.jpeg"
        />
        <div
          class="description-container-3"
        >
          <div
            class="key-word-3"
          >
            INTERACTIVE LESSONS
          </div>
          <div
            class="title-3"
          >
            Engage in dynamic learning
          </div>
          <div
            class="description-3"
          >
            Experience interactive lessons that keep you engaged and motivated. Immerse yourself in multimedia content, including audio, video, and interactive exercises. 
            <br />
            <br />
            Practice speaking and listening skills with speech recognition technology. Receive instant feedback and personalized recommendations for improvement.
          </div>
        </div>
      </div>
      <div
        class="main-content-4"
      >
        <div
          class="description-container-4"
        >
          <div
            class="key-word-4"
          >
            CULTURAL IMMERSION
          </div>
          <div
            class="title-4"
          >
            Explore language and culture
          </div>
          <div
            class="description-4"
          >
            Immerse yourself in the richness of different cultures through authentic content. Discover customs, traditions, and cultural nuances while learning the language.
            <br />
            <br />
             Engage in dialogues and scenarios that reflect real-life cultural situations and gain a deeper understanding of the language's cultural context.
          </div>
        </div>
        <img
          alt="japan at night"
          class="japan"
          src="/images/japan-night.jpeg"
        />
      </div>
      <div
        class="main-content-5"
      >
        <div
          class="description-container-5"
        >
          <div
            class="key-word-5"
          >
            Choose your language
          </div>
          <div
            class="description-5"
          >
            Unlock the world, discover your language journey
          </div>
          <a
            href="/shopping"
          >
            <button
              class="shop-2"
            >
              <span>
                Shop
              </span>
            </button>
          </a>
        </div>
      </div>
      <div
        class="main-content-6"
      >
        <div>
          Have a question?
        </div>
        <div
          class="cards-container"
        >
          <div
            class="help-center"
          >
            <div
              class="help-center-text"
            >
              <div
                class="help-center-text-title"
              >
                Go to Help Center
              </div>
              <div
                class="help-center-text-description"
              >
                Find answers and suggestions to most common questions
              </div>
            </div>
            <img
              alt="help center icon"
              src="/images/help-center.png"
            />
          </div>
          <div
            class="community"
          >
            <div
              class="community-text"
            >
              <div
                class="community-text-title"
              >
                Ask the community
              </div>
              <div
                class="community-text-description"
              >
                Create connections with music creators like you
              </div>
            </div>
            <img
              alt="community icon"
              src="/images/community.png"
            />
          </div>
        </div>
      </div>
      <div
        class="info"
      >
        <div
          class="logo"
        >
          <div
            class="logo-wrapper"
          >
            <img
              alt="earth logo"
              src="/images/earth-logo.png"
            />
            <span>
              roamingterra
            </span>
          </div>
        </div>
        <div
          class="company-info-container"
        >
          <div>
            Company
          </div>
          <div>
            About us
          </div>
          <div>
            People
          </div>
          <div>
            Brand Resources
          </div>
          <div>
            Press
          </div>
          <div>
            Contact
          </div>
          <div>
            Blog
          </div>
          <div>
            Podcast
          </div>
        </div>
        <div
          class="products-info-container"
        >
          <div>
            Products
          </div>
          <div>
            For Teachers
          </div>
          <div>
            For Schools
          </div>
          <div>
            For Partners
          </div>
          <div>
            For Developers
          </div>
          <div>
            For the Community
          </div>
        </div>
        <div
          class="community-info-container"
        >
          <div>
            Community
          </div>
          <div>
            Overview
          </div>
          <div>
            Guidelines
          </div>
          <div>
            Become a Curator
          </div>
          <div>
            Help Center
          </div>
          <div>
            Ask the Community
          </div>
          <div>
            Community Blog
          </div>
        </div>
      </div>
    </div>
    <div
      class="footer"
    >
      <div
        class="logo-small"
      >
        roamingterra
      </div>
      <div
        class="legal-information"
      >
        <div>
          Privacy Policy
        </div>
        <div>
          Cookie Policy
        </div>
        <div>
          EULA
        </div>
        <div>
          Copyright
        </div>
      </div>
      <div
        class="credits"
      >
        <div>
          Website created by Daniel Ballerini
        </div>
        <div>
          with help from 
          <a
            href="https://www.theodinproject.com/"
          >
             Â© Odin Project
          </a>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
